src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.c

```diff
@@ -33,42 +38,689 @@ const ActorInit Object_Kankyo_InitVars = {
+        D_801F4E30 = 155.0f * temp_f0;
+        globalCtx->envCtx.unk_EA = 10;
+    } else {
+        D_801F4E30 = 0;
+        globalCtx->envCtx.unk_EA = 10;
+    }
+    func_808DC454(this, globalCtx);
+}
 
-extern UNK_TYPE D_01000000;
+void func_808DCDB4(ObjectKankyo* this, GlobalContext* globalCtx) {
+    s16 i;
+    f32 sqrt;
+    f32 temp_80;
+    f32 temp_120;
+    f32 spAC;
+    f32 spA8;
+    f32 spA4;
+    f32 spA0;
+    f32 sp9C;
+    f32 x;
+    f32 y;
+    f32 z;
+    f32 temp_f18;
+    f32 temp_f20;
+    f32 temp_f26;
+    f32 temp_f28;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Object_Kankyo/ObjectKankyo_SetupAction.s")
+    if (this->unk_114C < 0x80) {
+        this->unk_114C++;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Object_Kankyo/func_808DBE8C.s")
+    x = globalCtx->view.at.x - globalCtx->view.eye.x;
+    y = globalCtx->view.at.y - globalCtx->view.eye.y;
+    z = globalCtx->view.at.z - globalCtx->view.eye.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Object_Kankyo/func_808DBEB0.s")
+    sqrt = sqrtf(SQ(x) + SQ(y) + SQ(z));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Object_Kankyo/func_808DBFB0.s")
+    spAC = x / sqrt;
+    spA8 = y / sqrt;
+    spA4 = z / sqrt;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Object_Kankyo/func_808DC038.s")
+    temp_80 = 80.0f;
+    temp_120 = 120.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Object_Kankyo/ObjectKankyo_Init.s")
+    for (i = 0; i < this->unk_114C; i++) {
+        switch (this->unk_14C[i].unk_1C) {
+            case 0:
+                this->unk_14C[i].unk_00 = globalCtx->view.eye.x + (spAC * 120.0f);
+                this->unk_14C[i].unk_04 = globalCtx->view.eye.y + (spA8 * 120.0f);
+                this->unk_14C[i].unk_08 = globalCtx->view.eye.z + (spA4 * 120.0f);
+                this->unk_14C[i].unk_0C = (Rand_ZeroOne() - 0.5f) * (temp_120 * 2.0f);
+                if ((i & 1) == 0) {
+                    this->unk_14C[i].unk_10 = -100.0f;
+                } else {
+                    this->unk_14C[i].unk_10 = 100.0f;
+                }
+                this->unk_14C[i].unk_14 = (Rand_ZeroOne() - 0.5f) * (temp_120 * 2.0f);
+                this->unk_14C[i].unk_18 = Rand_ZeroOne() + 0.2f;
+                this->unk_14C[i].unk_1C++;
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Object_Kankyo/ObjectKankyo_Destroy.s")
+            case 1:
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Object_Kankyo/func_808DC18C.s")
+                temp_f26 = globalCtx->view.eye.x + (spAC * 120.0f);
+                temp_f28 = globalCtx->view.eye.y + (spA8 * 120.0f);
+                temp_f18 = globalCtx->view.eye.z + (spA4 * 120.0f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Object_Kankyo/func_808DC454.s")
+                sqrt = sqrtf((f32)SQ(globalCtx->envCtx.windDir.x) + SQ(globalCtx->envCtx.windDir.y) +
+                             SQ(globalCtx->envCtx.windDir.z));
```

Comment:
```suggestion
                magnitude = sqrtf(SQXYZ(globalCtx->envCtx.windDir.x));
```

Reply:
Doesn't work due to losing the float cast.